ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -513,14 +563,7 @@ let update_allowed_operations ~__context ~self =
     | None -> op :: accu
     | _    -> accu
   in
-  let allowed =
-    List.fold_left check []
-      [`snapshot; `copy; `clone; `revert; `checkpoint; `snapshot_with_quiesce;
-       `start; `start_on; `pause; `unpause; `clean_shutdown; `clean_reboot;
-       `hard_shutdown; `hard_reboot; `suspend; `resume; `resume_on; `export; `destroy;
-       `provision; `changing_VCPUs_live; `pool_migrate; `migrate_send; `make_into_template; `changing_static_range;
-       `changing_shadow_memory; `changing_dynamic_range]
```

Comment:
Hmm, I can't see any expensive check in `Xapi_vm_lifecycle.check_operation_error` that is activated by passing to it a specific operation. I talked to this about @jonludlam and he said that we can just extend it to the full set, to achieve a more correct behaviour, but it might cause bugs. But I'll remove this change if it's a deliberate choice.